src/index/index.c

```diff
@@ -19,13 +20,24 @@
 void Index_RangeFieldName
 (
 	char *type_aware_name,  // [out] type aware name
-	const char *name        // field name
+	const char *name,       // field name
+	SIType *multi_val_type  // [optional] multi-val type
 ) {
-	ASSERT(name != NULL);
+	ASSERT(name            != NULL);
 	ASSERT(type_aware_name != NULL);
-
-	// prefix range field name with "range:"
-	sprintf(type_aware_name, "range:%s", name);
+	ASSERT(multi_val_type  == NULL ||
+		   *multi_val_type & (T_STRING | SI_NUMERIC | T_BOOL));
+
+	if(unlikely(multi_val_type != NULL)) {
+		if(*multi_val_type == T_STRING) {
+			sprintf(type_aware_name, "range:%s:string:arr", name);
```

Comment:
type_aware_name is char type_aware_field_name[512]; do we have limit on the field name ?

Reply:
We don't I'll assume 512 should be enough, I know this is optimistic thinking...